docker/build-project.sh

```diff
@@ -25,6 +25,12 @@ cd $WORKSPACE
 # get git commit ID of the project for using in buildhistory tagging
 CI_GIT_COMMIT=$(git rev-parse HEAD)
 
+# prepare git config
+#NOTE: bitbake does not pass git enviornment to tasks(do_patch),
+# hence we need to setup the git configuration 
+git config --global user.name "$GIT_COMMITTER_NAME"
+git config --global user.email "$GIT_COMMITTER_EMAIL"
+
 # FIXME: undbound variables used without checking:
```

Comment:
@Kad, This is not just related to git apply or patch formatting, As i mentioned in commit message the real issue is with the git environment in bitbake tasks, though we set GIT_COMITTER_{NAME/EMAIL},  those are not available to while patching source. So here the idea is to pass them via 'git config'